OpenRA.Mods.Common/TargetExtensions.cs

```diff
@@ -42,7 +42,18 @@ public static Target Recalculate(this Target t, Player viewer, out bool targetIs
 
 			// Bot-controlled units aren't yet capable of understanding visibility changes
 			if (viewer.IsBot)
+			{
+				if (t.Type == TargetType.FrozenActor)
+				{
+					if (t.FrozenActor.Actor != null)
+						return Target.FromActor(t.FrozenActor.Actor);
+
+					if (t.Actor == null)
+						return Target.Invalid;
```

Comment:
I copied this from below but don't know it's needed.

Reply:
Its probably a good idea to include it just to make sure we don't end up with a null pointer.

Reply:
If the `TargetType` is `FrozenActor`, then `Actor` is always null.

https://github.com/OpenRA/OpenRA/blob/2c24a607bc83e7929bdd373090ffddcbb769242d/OpenRA.Game/Traits/Target.cs#L79

This is unconditionally true and thereby always returns null.

Reply:
The fix doesn't work without it. We need to invalidate the target here.

Reply:
That's not my point.

You're introducing dead code here, because the code never reaches the `return t;` statement below, nor the if is required.

```
if (viewer.IsBot)
{
	if (t.Type == TargetType.FrozenActor)
		return Target.FromActor(t.FrozenActor.Actor);

	return Target.Invalid;
}

Reply:
Ok, thanks for explaining this again. Though I don't understand why ` the code never reaches the return t; statement below` - it should do so for `TargetType.Actor`. And we don't want to go through the normal visibility checks for bots

Reply:
Ah, yeah, I mixed up the levels of indentation. The if is still redundant however.